motoman_sia50_support/urdf/sia50d_macro.xacro

```diff
@@ -0,0 +1,209 @@
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_u">
+            <visual>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_4_u.stl"/>
+                </geometry>
+                <material name="blue"/>
+            </visual>
+            <collision>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_4_u.stl"/>
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_r">
+            <visual>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_5_r.stl"/>
+                </geometry>
+                <material name="silver"/>
+            </visual>
+            <collision>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_5_r.stl"/>
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_b">
+            <visual>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_6_b.stl"/>
+                </geometry>
+                <material name="blue"/>
+            </visual>
+            <collision>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_6_b.stl"/>
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_t">
+            <visual>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_7_t.stl" />
+                </geometry>
+                <material name="silver"/>
+            </visual>
+            <collision>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_7_t.stl" />
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}tool0" />
+        <!-- end of link list -->
+        <!-- joint list -->
+        <joint name="${prefix}joint_s" type="revolute">
+            <parent link="${prefix}base_link"/>
+            <child link="${prefix}link_s"/>
+            <origin xyz="0 0 0.312000" rpy="0 0 0"/>
```

Comment:
Is there a specific reason you have the Z translation in 6 significant digits here?